src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/BlockCollection.js

```diff
@@ -37,16 +37,26 @@ class BlockCollection extends React.Component<Props> {
         super(props);
 
         this.fillArrays();
+        reaction(() => this.props.value.length, this.fillArrays.bind(this));
     }
 
     fillArrays() {
         const {defaultType, onChange, minOccurs, value} = this.props;
         const {expandedBlocks, generatedBlockIds} = this;
 
-        if (!value) {
+        if ((!value || expandedBlocks.length === value.length || generatedBlockIds.length === value.length)
+                && value.length !== 0) {
             return;
         }
 
+        if (expandedBlocks.length > value.length) {
+            expandedBlocks.length = 0;
+        }
+
+        if (generatedBlockIds.length > value.length) {
+            generatedBlockIds.length = 0;
+        }
```

Comment:
I don't really get why these two `if`s are necessary... Can you elaborate?

Reply:
Those are necessary for the case, when at first we have e.g. 5 `blocks` in the `value` array, thus `generatedBlockIds` and `expandedBlocks`, all 3 arrays have a length of 5

then we change the data in `value` e.g. through a toolbarAction to only 2 `blocks` and now we also have to adjust the `expandedBlocks` and `generatedBlockIds` otherwise we would run into an error in the following lines

`expandedBlocks.push(...new Array(value.length - expandedBlocks.length).fill(false));` and 
`        generatedBlockIds.push(
            ...new Array(value.length - generatedBlockIds.length).fill(false).map(() => ++BlockCollection.idCounter)
        );`

Reply:
Wouldn't it be better to handle this case differently? Just sketching something for only the `expandedBlocks` array:

```javascript
if (expandedBlocks.length > value.length) {
    expandedBlocks.splice(value.length);
}
```

This would have the advantage, that the old values stay. As I see it, the `reaction` you've wrote is also fired when a new block is added using the button, and then all blocks would have to collapse. Have you tested that?

Reply:
I guess you are right, this would also work. I will change it later.

Yes I have tested it.
If it is fired when a block is added or deleted, this condition is true 
```
 if ((!value || expandedBlocks.length === value.length || generatedBlockIds.length === value.length)
                && value.length !== 0) {
            return;	            return;
        }	        
}
```

because `expandedBlocks` and `generatedBlockIds` were already updated during the `add block` method.

Reply:
Ah I see... That was not obvious on the first look. I think the other way (without making the initial `if` with the early return more complicated) would be easier to understand, do you agree? Plus it would have the advantage that the values stay the same, which might not be important in your use case, but maybe in a different one.

Reply:
Yes I agree, fixed it :)